### PR TITLE
Applied `@abc.abstractmethod` decorator to the abstract methods of `BaseTrial`

### PR DESCRIPTION
**Pull request by gorogoroumaru**
issue #1084

This PR applied `@abc.abstractmethod` decorator to the abstract methods of `BaseTrial`

### CODE REVIEW

File: optuna/integration/chainermn.py [hunk 1]
```diff
@@ -189,6 +189,17 @@ def __init__(self, trial, comm):
...
         self.comm = comm
 
+    def suggest_float(self, name, low, high, *, log=False):
+        # type: (str, float, float, bool) -> float
+
+        def func():
+            # type: () -> float
+
+            assert self.delegate is not None
+            return self.delegate.suggest_float(name, low, high)
```
Comment (crcrpar) [inline optuna/integration/chainermn.py:199]: It seems like the keyword argument of `log` is ignored.

Isn't it better to pass `log` to `self.delegate.suggest_float`?

Comment (crcrpar) [inline optuna/integration/chainermn.py:199]: It seems like the keyword argument of `log` is ignored.

Isn't it better to pass `log` to `self.delegate.suggest_float`?

Comment (gorogoroumaru) [reply optuna/integration/chainermn.py:199]: Thank you!
I will fix it.

**Comment by crcrpar** [comment]
Thanks for the swift update. In my opinion, the newly implemented `suggest_float` should also be tested (ref: https://github.com/optuna/optuna/pull/1021) so that now I think this PR has better to be separated.

However, again, I'm not sure, so you don't need to hurry to separate this PR. 🙂

File: tests/integration_tests/test_chainermn.py [hunk 2]
```diff
@@ -304,6 +304,37 @@ def test_init(storage_mode, comm):
 
             assert mn_trial.number == trial.number
 
+    @staticmethod
+    @pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+    def test_suggest_float(storage_mode, comm):
```
Comment (hvy) [inline tests/integration_tests/test_chainermn.py:309]: How about also checking that exceptions are raised as excepted similar to other tests below?

Comment (gorogoroumaru) [reply tests/integration_tests/test_chainermn.py:309]: Thanks!
I applied your suggested change.

Comment (hvy) [reply tests/integration_tests/test_chainermn.py:309]: Thanks for your quick action.

**Comment by hvy** [comment]
Allow me to update the title to somewhat better reflect the changes.

**Pull request merged**